securedrop/journalist.py

```diff
@@ -593,11 +602,24 @@ def reply():
                         [crypto_util.getkey(g.sid), config.JOURNALIST_KEY],
                         output=store.path(g.sid, filename))
     reply = Reply(g.user, g.source, filename)
-    db_session.add(reply)
-    db_session.commit()
 
-    flash("Thanks! Your reply has been stored.", "notification")
-    return redirect(url_for('col', sid=g.sid))
+    try:
+        db_session.add(reply)
+        db_session.commit()
+    except Exception as exc:
+        flash("An unexpected error occurred! Please check the application "
+              "logs or inform your adminstrator.", "error")
+        # We take a cautious approach to logging here because we're dealing
+        # with responses to sources. It's possible the exception message could
+        # contain information we don't want to write to disk.
+        app.logger.error(
+            "Reply from '{}' (id {}) failed: {}!".format(g.user.username,
+                                                         g.user.id,
+                                                         exc.__class__))
+    else:
+        flash("Thanks! Your reply has been stored.", "notification")
+    finally:
+        return redirect(url_for('col', sid=g.sid))
```

Comment:
Question: is this PR intended to handle the case where the journalist tries to reply to a recently deleted source? It looks like from the use of the `StaleDataError` exception in the unit tests that the answer is yes - but in this case, this redirect to `url_for('col', sid=g.sid)`, will 404 if the source was deleted, so the flashed message above will not be displayed, no?